src/detailed/authentication_and_encryption.md

```diff
@@ -0,0 +1,48 @@
+# Authentication & Encryption
+This page details how Authentication & Encryption may work for Eludris.
+
+## The Token
+
+Eludris uses JWT tokens to authenticate users. These tokens are required for nearly every interaction. Connection to the Gateway or trying to interact with the API? You'll need a token!
+The JWT tokens used within our services use a cryptographically secure pseudo-random string with HC128 as a secret. 
+If you wish to get a new token send an HTTP request to `/auth` with your email and password.
+
+## End-To-End-Encryption
+
+End-To-End-Encryption (we'll be calling it E2EE for now on) will be available to private communities, private GDMs (group direct messages) and direct messages (DMs) between friends.
+
+### E2EE Implementation
+
+To introduce this, every user has a personal, unique, public and private key provided.
+
+Events with encrypted data (message, post, etc) have an extra field in their payload, the `pubkey` field which contains the public key the message's content was encrypted with so that the corresponding private key would be fetched from the user's public-private key pairs and requested if the current one is invalid.
+
+As for storing public-private key pairs, storing them locally (on the client's machine) causes a lot of extra complexity, especially with sharing and syncing keys.
+
+For example, issues with a client being offline when it's given a key, multiple clients and so on.
+
+To combat that, Eludris' E2EE is designed so that each user has a super private-public key pair that their other private keys are encrypted with.
+
+The instance *does not know* the user's super private key, additionally the instance gives the user all the (unencrypted) public and (ecnrypted with each user's super public key) private keys on connecting to Pandemonium, the instance ***never*** gets access to the non-encrypted private keys of *any* key pair at any point in time.
```

Comment:
~~ecnrypted~~ -> encrypted

Also, I feel like this could be split into multiple sentences.